Update and Extend Documentation

markulf opened the pull request:
Starting with broken links. But any other kind of documentation is also welcome. 
Might by its very nature, documentation is never finished, be broken up into multiple pull requests.

markulf committed:
Added CHANGELOG with some examples from closed pull requests

markulf committed:
Added CHANGELOG with some examples from closed pull requests

markulf committed:
Added CHANGELOG with some examples from closed pull requests

markulf committed:
precise pointer to documentation index.html

markulf committed:
precise pointer to documentation index.html

markulf committed:
precise pointer to documentation index.html

markulf committed:
Fixed broken links to be compatible with includes in plonk-core/src.rs

markulf committed:
Fixed broken links to be compatible with includes in plonk-core/src.rs

markulf committed:
added useful pointer to keepachangelog and comment that this only starts from version 0.1

markulf committed:
added useful pointer to keepachangelog and comment that this only starts from version 0.1

markulf committed:
added useful pointer to keepachangelog and comment that this only starts from version 0.1

markulf committed:
Added a simple example

markulf committed:
Added a simple example

markulf reviewed (commented):
The title of 3rd commit should be plonk-core/src/lib.rs

Likely these includes should just be removed and the files deleted.

markulf committed:
Just simple editorial stuff for comments

markulf committed:
Just simple editorial stuff for comments

CPerezz commented on examples/simple_circuit.rs:
```diff
@@ -0,0 +1,171 @@
+    &pi_pos,
+    b"Test",
+)
+}
```
Add new line at the end of the file

CPerezz commented on examples/simple_circuit.rs:
```diff
@@ -0,0 +1,171 @@
+//! PLONK Example
+
+use ark_bls12_381::{Bls12_381, Fr as BlsScalar};
+//use ark_ec::{PairingEngine, TEModelParameters};
```
Remove the commented stuff

CPerezz reviewed (commented):
Looks good. 

We might add a bit simpler examples IMO. Or group them by:
- Arith
- Bool
- ECC
- Range
- Misc

markulf committed:
added final linebreak

markulf committed:
Merge branch 'improve_doc' of github.com:ZK-Garage/plonk into improve_doc

markulf committed:
cleanup and regression adjustment for PublicInputBuilder

LukePearson1 committed:
Merge branch 'improve_doc' of https://github.com/rust-zkp/ark-plonk into improve_doc

LukePearson1 committed:
Merge branch 'master' into improve_doc

LukePearson1 committed:
Apply cargo fmt

LukePearson1 committed:
Refactor circuit example

LukePearson1 reviewed (approved):
LGTM! 

Had to so some merging to get up to date with the latest branch.

LukePearson1 committed:
Remove serialisation

The serialisation method is removed
from example circuit as the whole
code block was redundant.

The serialisation tests are done
in the plonk-core module.

LukePearson1 committed:
Remove serialisation deps

bhgomes commented on examples/simple_circuit.rs:
```diff
@@ -0,0 +1,154 @@
+use plonk_core::constraint_system::StandardComposer;
+use plonk_core::error::Error;
+
+use ark_serialize::{CanonicalDeserialize, CanonicalSerialize};
```
@LukePearson1 still need to remove this line

LukePearson1 committed:
Apply rust fmt
